src/HttpAsyncClientDiscovery.php

```diff
@@ -24,7 +24,7 @@ public static function find()
         try {
             $asyncClient = static::findOneByType(HttpAsyncClient::class);
 
-            return new $asyncClient();
+            return static::instantiateClass($asyncClient);
```

Comment:
Do we need the instantiation within the try? Is it possible that a closure throws a DiscoveryFailedException?